repo/grpc_repository_client_unit_test.go

```diff
@@ -0,0 +1,64 @@
+package repo
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestBaseURLToURI(t *testing.T) {
+	for _, tc := range []struct {
+		name      string
+		baseURL   string
+		expURI    string
+		expErrMsg string
+	}{
+		{
+			name:      "ipv4",
+			baseURL:   "https://1.2.3.4:5678",
+			expURI:    "1.2.3.4:5678",
+			expErrMsg: "",
+		},
+		{
+			name:      "ipv6",
```

Comment:
nit; I don't think this parsing function is sensitive to link-local addressing (fe80::/10 namespace), but why not add a test case anyway since now you have it ?

Example link local addr: fe80::d6be:d9ff:feb0:562d

Disclaimer: I don't even know if rest of Kopia works with link-local addresses.